docs/usage/with-typescript.md

```diff
@@ -0,0 +1,224 @@
+---
+id: with-typescript
+title: Usage With TypeScript
+sidebar_label: Usage With TypeScript
+hide_title: true
+---
+
+# Usage With TypeScript
+
+It is one of Redux Toolkit's design targets to offer great integration with TypeScript.
+
+This page is intended to give an overview of all common usecases and the most probable pitfalls you might encounter when using RTK with TypeScript.
+
+**If you encounter any problems with the Types that are not described on this page, please open an Issue!**
+
+## Using configureStore with TypeScript
+
+Using [configureStore](../api/configureStore.ms) should not need any additional typings. You might, however, want to extract the `RootState` type and the `Dispatch` type.
+
+### Getting the `State` type
+
+The easiest way of getting the `State` type is to define the root reducer in advance and extract it's `ReturnType`.  
+It's recommend to give it a different name like `RootState` to prevent confusion, as the type name `State` is usually overused.
+
+```typescript
+import { combineReducers } from '@reduxjs/toolkit'
+const rootReducer = combineReducers({})
+export type RootState = ReturnType<typeof rootReducer>
+```
+
+### Getting the `Dispatch` type
+
+If you want to get the `Dispatch` type from your store, you can extract it after creating the store.  
+It's recommend to give it a different name like `AppDispatch` to prevent confusion, as the type name `Dispatch` is usually overused.
+
+```typescript
+import { configureStore } from '@reduxjs/toolkit'
+import rootReducer from './rootReducer'
+const store = configureStore({
+  reducer: rootReducer
+})
+export type AppDispatch = typeof store.dispatch
+```
+
+### Extending the `Dispatch` type
+
+By default, this `AppDispatch` type will account only for the already included `redux-thunk` middleware. If you're adding additional middlewares that provide different return types for some actions, you can overload that `AppDispatch` type. While you can just extend the `AppDispatch` type, it's recommended to do so on the store, to keep everything more consistent:
+
+```typescript
+const _store = configureStore({
+  /* ... */
+})
+
+type EnhancedStoreType = {
+  dispatch(action: MyCustomActionType): MyCustomReturnType
+  dispatch(action: MyCustomActionType2): MyCustomReturnType2
+} & typeof _store
+
+export const store: EnhancedStoreType = _store
+export type AppDispatch = typeof store.dispatch
+```
+
+### Using the extracted `Dispatch` type with react-redux
```

Comment:
Capitalize "React-Redux" everywhere.  (Yeah, I know, we're really inconsistent about capitalization and naming of things like "Redux Thunk' vs `redux-thunk`, etc.

Reply:
Noted :)